cppcache/src/ClientMetadata.cpp

```diff
@@ -114,14 +114,24 @@ const std::string& ClientMetadata::getColocatedWith() {
   return m_colocatedWith;
 }
 
+void ClientMetadata::removeBucketServerLocation(
+    const std::shared_ptr<BucketServerLocation>& serverLocation) {
+  for (auto&& locations : m_bucketServerLocationsList) {
+    for (unsigned int i = 0; i < locations.size(); i++) {
+      if (locations[i]->getEpString() == (serverLocation->getEpString())) {
+        locations.erase(locations.begin() + i);
```

Comment:
Won't erasing an item from `locations` make it invalid in the `for` loop's `locations.size()` ?

Reply:
Yes it will but given that when an element is erased we right after exit the loop with the break sentence then it does not matter.

Reply:
Oh yes, there is a `break` statement, I am blind!